Framework/TestHelpers/src/ComponentCreationHelper.cpp

```diff
@@ -603,12 +603,99 @@ Instrument_sptr createTestInstrumentRectangular(int num_banks, int pixels,
  * Pixels are 4 mm wide.
  *
  * @param num_banks: number of rectangular banks to create
- * @param pixels :: number of pixels in each direction.
- * @param pixelSpacing :: padding between pixels
+ * @param pixels : number of pixels in each direction.
+ * @param nameless : components are unnamed if true.
+ * @param pixelSpacing : padding between pixels
  */
 Instrument_sptr createTestInstrumentRectangular2(int num_banks, int pixels,
+                                                 bool nameless,
                                                  double pixelSpacing) {
-  auto testInst = boost::make_shared<Instrument>("basic_rect");
+
+  auto instrName = nameless ? "" : "basic_rect";
+  auto testInst = boost::make_shared<Instrument>(instrName);
+
+  const double cylRadius(pixelSpacing / 2);
+  const double cylHeight(0.0002);
+  // One object
+  auto pixelShape = ComponentCreationHelper::createCappedCylinder(
+      cylRadius, cylHeight, V3D(0.0, -cylHeight / 2.0, 0.0), V3D(0., 1.0, 0.),
+      "pixel-shape");
+
+  if (nameless) {
+    for (int banknum = 1; banknum <= num_banks; banknum++) {
+      // Make a new bank
+      std::ostringstream bankname;
+      bankname << "";
+
+      RectangularDetector *bank = new RectangularDetector(bankname.str());
```

Comment:
Use `auto` to avoid type name repetition.

Reply:
done